### DISCUSSION TITLE
Retain notification topics' subscribers in major update

### DISCUSSION THREAD

**Discussion by klaevv** [invertase/react-native-firebase#4414] [Q&A]
# Context

Our app has users who are subscribed to various notification topics via react-native-firebase. We are planning on a major update and wish to retain the existing subscriptions for each topic.

# Question

If we rewrite our whole codebase but keep the old bundle ID (iOS) and applicationId (Android), will our existing users retain their subscriptions for topics if we release the new app version as an update?

> An Instance ID is long lived except in a few scenarios:
> 
> * when you call delete()
> * the app is restored on a new device
> * the user uninstalls/reinstall the app
> * [android] the user clears the app data

The above is quoted from [here](https://developer.aliyun.com/mirror/npm/package/@react-native-firebase/iid/v/6.7.0). It looks like the subscriptions are retained as long as the new version is served as an update, but we'd like to be sure.

**Comment by mikehardy** [comment]
In my opinion, this is too important for the success of an app to leave to any chance. I'd test it with a separate testing bundle ID

**Reply by klaevv** [reply]
OK, we'll test it with a separate bundle ID 👍 My hope is that the instance ID would be persisted through the update.

I wonder has anyone else had a similar case...
